src/components/storefront/ProductDetails/ProductDetails.js

```diff
@@ -50,13 +50,26 @@ const ProductDetails = ({ product, checkout }) => {
           {`$${variantPrice}`}
         </Heading>
         <Text my={5}>{description}</Text>
-        <QuantityPicker {...quantityPickerProps} />
-        <Button textTransform="uppercase" mt={5} mb={10} onClick={handleAddToShoppingBag}>
-          {'Add to shopping bag'}
+        {availableForSale ? (
+          <QuantityPicker {...quantityPickerProps} />
+        ) : (
+          <Heading as="h3" size="h3" textTransform="uppercase">
+            Out of Stock
+          </Heading>
+        )}
+        <Button
+          textTransform="uppercase"
+          mt={5}
+          mb={10}
+          minW="auto"
+          onClick={handleAddToShoppingBag}
+          disabled={!availableForSale}
+        >
```

Comment:
On hover, the button turns white so it actually disappears. I'm not sure if that's intended but I think a disabled button with no hover effect is generally how Out of Stock buttons look.

Reply:
Great catch, yeah this is an annoyance in chakra UI (maybe an oversight in the library).

Fixed in a75bd63